### PR TITLE
add soci 4.0.2 package

### PR DESCRIPTION
**Pull request by spjuanjoc**
Library name and version:  **soci/4.0.2**

Add the recently released version 4.0.2 to the existing SOCI package.
---

- [X] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [X] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [X] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [X] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/soci/all/conanfile.py [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import os
+import shutil
```
Comment (prince-chrismc) [inline recipes/soci/all/conanfile.py:2]:
```suggestion
```

File: recipes/soci/all/conanfile.py [hunk 2]
```diff
@@ -151,6 +152,7 @@ def package(self):
                 tools.rename(os.path.join(self.package_folder, "lib64"), os.path.join(self.package_folder, "lib"))
 
         os.remove(os.path.join(self.package_folder, "include", "soci", "soci-config.h.in"))
+        shutil.rmtree(os.path.join(self.package_folder, "lib", "cmake"), ignore_errors=True)
```
Comment (prince-chrismc) [inline recipes/soci/all/conanfile.py:155]:
```suggestion
        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
```

**Review by prince-chrismc — CHANGES_REQUESTED** [review]
Please try to use the conan tools

https://docs.conan.io/en/latest/reference/tools.html#tools-mkdir-tools-rmdir

**Comment by prince-chrismc** [comment]
For anyone reviewing after... I did not see any components, its just vendored module files that I found

**Review by SSE4 — APPROVED** [review]

**Review by ericLemanissier — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Pull request merged**